vsintegration/src/FSharp.Editor/Structure/BlockStructureService.fs

```diff
@@ -148,6 +148,8 @@ type internal FSharpBlockStructureService(checker: FSharpChecker, projectInfoMan
  
     override __.GetBlockStructureAsync(document, cancellationToken) : Task<BlockStructure> =
         asyncMaybe {
+            do! Option.guard Settings.Advanced.IsBlockStructureEnabled
```

Comment:
This code basically means "only proceed to get the info for block structure if that setting has a value of `true`"